components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java

```diff
@@ -167,6 +171,16 @@ protected void activate(ComponentContext ctxt) {
                 handleFatal("Couldn't register the SynapseConfigurationService, " + "SynapseConfiguration not found");
             }
             SynapseEnvironment synapseEnvironment = contextInfo.getSynapseEnvironment();
+            List handlers = synapseEnvironment.getSynapseHandlers();
+            Iterator<SynapseHandler> iterator = handlers.iterator();
+            while (iterator.hasNext()) {
+                SynapseHandler handler = iterator.next();
+                if ((handler instanceof AbstractExtendedSynapseHandler)) {
+                    if (!((AbstractExtendedSynapseHandler) handler).handleServerInit()) {
+                        return;
```

Comment:
Why are we returning from the method here? Should the server should not be started if handler serverinit fails?

Reply:
Logged the warning